src/framework/tooltip/Tooltip.jsx

```diff
@@ -23,65 +25,53 @@ export default class Tooltip extends Component {
       this.content.style.width = this.props.width;
       const styles = ['tooltip__container'];
 
-      let isPosXSet = false;
-      let isPosYSet = false;
+      let isRenderedToRight = false;
+      let isRenderedToTop = false;
 
-      const tooltipDimension = this.getTooltipDimension();
-      let tooltipX =
-        (this.tooltip.clientWidth / 2) - (tooltipDimension.width - 21);
-      if (tooltipDimension.posX - tooltipDimension.width < 0) {
-        tooltipX = (this.tooltip.clientWidth / 2) - 18;
-        styles.push('tooltipRight');
-        isPosXSet = true;
+      // Adjust position to the right if window requires.
+      this.setPosX(this.getTooltipXToLeft());
```

Comment:
Setting position X (`this.container.style.left`) before checking the position, makes the check easier.